src/main/groovy/org/shipkit/internal/gradle/git/GitAuthPlugin.java

```diff
@@ -0,0 +1,113 @@
+package org.shipkit.internal.gradle.git;
+
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
+import org.shipkit.gradle.ReleaseConfiguration;
+import org.shipkit.internal.exec.DefaultProcessRunner;
+import org.shipkit.internal.gradle.configuration.ReleaseConfigurationPlugin;
+import org.shipkit.internal.notes.vcs.GitOriginRepoProvider;
+
+import java.text.MessageFormat;
+
+public class GitAuthPlugin implements Plugin<Project>{
+
+    private static final Logger LOG = Logging.getLogger(GitAuthPlugin.class);
+
+    private GitAuth gitAuth;
+
+    @Override
+    public void apply(Project project) {
+        ReleaseConfiguration conf = project.getPlugins().apply(ReleaseConfigurationPlugin.class).getConfiguration();
+        String ghUser = conf.getGitHub().getWriteAuthUser();
+        String writeToken = conf.getGitHub().getWriteAuthToken();
+
+        String configUrl;
+        String originUrl;
+        String secretValue = null;
+        String originRepositoryName = getOriginRepositoryName(project);
```

Comment:
This method triggers an external process so we need to do it in the execution time.

There is no easy way to model this in Gradle. I suggest to try out the approach I did with the IdentifyGitBranchTask. Perhaps a new task IdentifyGitOriginTask? If you find a better solution - let me know!